### DISCUSSION TITLE
[SOLVED] GitHub Copilot could not connect to server

### DISCUSSION THREAD

**Discussion by JorgeMGuimaraes** [community/community#10518] [Copilot Conversations]
I got this error message:
`[ERROR] [default] [2022-01-24T22:29:27.488Z] GitHub Copilot could not connect to server. Extension activation failed: "Cancelled"`

So I can`t activate. I already used and activated on another Ubuntu machine. 

Steps:
Click Accounts > Sign in with GitHub to use GitHub Copilot (1)
Click Signing in to Github.com... button on bottom bar
Pop up a blank text area (like ctrl+p) and a message:
`Uri (press 'Enter' to confirm or 'Escape' to cancel)`

There is no link to follow, just the upper message on console.

```
Version: 1.63.2
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:39:46.686Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.15.16-gentoo-dist
```

Edit:
[My solution](https://github.com/github/feedback/discussions/10518#discussioncomment-2069505)

**Comment by davecheney** [comment]
Could you please double check that you are running VSCode, not OSS Code or VSCodium.

**Reply by JorgeMGuimaraes** [reply]
Yes, I can confirm it´s VS Code, not OSS Code or VSCodium.

**Reply by davecheney** [reply]
Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by JorgeMGuimaraes** [reply]
No. No firewall installed and not using vpn. Is there any address I can ping to test?

**Reply by davecheney** [reply]
vscode-auth.github.com / api.github.com / copilot-proxy.githubusercontent.com

**Reply by subhobrata** [reply]
Hi All,
I was facing the same issue, then I closed the VS Code and again opened the VS code, then again reauthenticate with Github, then it worked

**Reply by JorgeMGuimaraes** [reply]
> vscode-auth.github.com / api.github.com / copilot-proxy.githubusercontent.com

@davecheney I can´t ping these addresses even on my machine where Copilot is working.

@subhobrata I tried this but didn´t work.

**Reply by davecheney** [reply]
are you sure your isp, employer, or government is not interfering with your internet connection?

**Reply by JorgeMGuimaraes** [reply]
Hi @davecheney , I fixed this issue by resseting my default browser (I dont know why this helped anyway):
```
user $ xdg-settings set default-web-browser firefox.desktop
user $ xdg-settings set default-url-scheme-handler http firefox.desktop
user $ xdg-settings set default-url-scheme-handler https firefox.desktop
``` 
This way, I clicked `Accounts > Sign in with GitHub to use GitHub Copilot (1)` and a new firefox window opened and I got an activation link. 

Note:
I still can´t ping `[vscode-auth.github.com / api.github.com / copilot-proxy.githubusercontent.com]` but successfully activated CoPilot.
As a susgestion: Copilot can print the login link to the VSCode output so if something like this happens the user has a plan B.

**Comment by redwine-1** [comment]
I was having the same problem. I just restarted vscode and it worked.

**Comment by Ezequie121** [comment]
Hello, I have this problem, how can I solve it? 
.- GitHub Copilot could not connect to server. Extension activation failed: "Cancelled"

**Reply by lioy98** [reply]
GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

**Reply by davecheney** [reply]
That error indicates you are still on the waitlist. I'm sorry that you have not been given access yet. Please see https://github.com/github/feedback/discussions/9342#discussioncomment-1902060

**Comment by Ezequie121** [comment]
Hello @davecheney , could you help me with this error please, I don't know what would be happening, I followed the steps perfectly and this error appears
"GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

**Reply by davecheney** [reply]
@Ezequie121 
This error indicates that you are either not part of the beta invite list, or there was an error connecting to api.github.com to request a token to use the Copilot service. Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by encoderitsajal** [reply]
[INFO] [auth] [2022-02-27T07:09:35.630Z] Invalid copilot token: missing token: 404 
[ERROR] [default] [2022-02-27T07:09:35.632Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

**Reply by encoderitsajal** [reply]
I get same error @davecheney

**Reply by davecheney** [reply]
That error indicates you are still on the waitlist.

**Reply by encoderitsajal** [reply]
@davecheney How long i get solve this error?

**Reply by davecheney** [reply]
I'm sorry that you have not been given access yet. Please see https://github.com/github/feedback/discussions/9342#discussioncomment-1902060

**Comment by rbarbrow** [comment]
I am having the same issue "GitHub Copilot could not connect to server. Extension activation failed: "Cancelled""  is there a port that this is using that may be blocked by my firewall?

**Reply by davecheney** [reply]
port 443

**Comment by PHOSPHENES8** [comment]
Hi there, I have met the same question and I found the point that the mistake lies in the setting sync on my VSCode--you have to make sure that you have signed in to your GitHub account (maybe you need to install setting sync first...). then you will see the tab window and you can see a pop-up window, which seems like: `GitHub Sign In`(p.s. translation may be a little different), just click it. Done. (Only for reference)

**Comment by riandroraul** [comment]
in vscode I had a similar error, what I did was to disable the extension > reload required > re-enable the github copilot extension and it worked fine

**Reply by davecheney** [reply]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.
